dkg/dkg.go

```diff
@@ -48,6 +48,52 @@ type Config struct {
 	TestSigning bool
 }
 
+type exchanger struct {
+	setChan chan core.ParSignedDataSet
+	sigex   *parsigex.ParSigEx
+}
+
+func newExchanger(tcpNode host.Host, peerIdx int, peers []peer.ID) exchanger {
+	ex := exchanger{
+		sigex:   parsigex.NewParSigEx(tcpNode, peerIdx, peers),
+		setChan: make(chan core.ParSignedDataSet, len(peers)),
+	}
+
+	// Wiring core workflow components
+	ex.sigex.Subscribe(ex.getSets)
+
+	return ex
+}
+
+func (e *exchanger) exchange(ctx context.Context, set core.ParSignedDataSet) ([]core.ParSignedDataSet, error) {
+	err := e.sigex.Broadcast(ctx, core.Duty{}, set)
+	if err != nil {
+		return nil, err
+	}
+
+	var sets []core.ParSignedDataSet
+	sets = append(sets, set)
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, err
+		case peerSet := <-e.setChan:
+			sets = append(sets, peerSet)
+		}
+		if len(sets) == cap(e.setChan) {
```

Comment:
we are done when we have ParSignedDataSet from all the peers including the current one

Reply:
nit: can we have the same as a comment?